dao/src/test/java/se/fortnox/reactivewizard/db/CollectionOptionsTest.java

```diff
@@ -73,12 +75,26 @@ void shouldNotSetLimitAboveMaxLimit() throws SQLException {
 
     @Test
     void shouldUseConfiguredDefaultLimit() throws SQLException {
+        mockDb.addRows(10);
         final CollectionOptions collectionOptions = new CollectionOptions();
-        collectionOptionsDao.selectWithDefaultLimit10(collectionOptions).collectList().block();
+        StepVerifier.create(collectionOptionsDao.selectWithDefaultLimit10(collectionOptions))
+            .expectNextCount(10)
+            .verifyComplete();
         mockDb.verifySelect("select * from table LIMIT 11");
         assertThat(collectionOptions.isLastRecord()).isTrue();
     }
 
+    @Test
+    void shouldUseConfiguredDefaultLimitOfOneWithReturnTypeMono() throws SQLException {
+        mockDb.addRows(1);
```

Comment:
I decided to cheat a bit here. The test fails when the num rows > 1 since MockDb does not look at the limit in the query. Having num rows > limit when the return type is Flux works fine because the PagingOperator filters out the remaining rows, but the PagingOperator is never used when working with Monos.

Reply:
That said, this has been tested against an actual postgres database and it works as expected.